src/Plugin/resource/DataProvider/DataProviderEntity.php

```diff
@@ -152,7 +150,20 @@ public function getIndexIds() {
       return array();
     }
 
-    return array_keys($result[$this->entityType]);
+    $entity_ids = array_keys($result[$this->entityType]);
+    if (empty($this->options['idField'])) {
+      return $entity_ids;
+    }
+
+    // Get the list of IDs.
+    $resource_field = $this->fieldDefinitions->get($this->options['idField']);
+    $ids = array();
+    foreach ($entity_ids as $entity_id) {
+      $interpreter = new DataInterpreterEMW($this->getAccount(), new \EntityDrupalWrapper($this->entityType, $entity_id));
```

Comment:
`entity_metadata_wrapper` yields `EntityDrupalWrapper` for entities. `\EntityMetadataWrapper` is not good enough in this case.